File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& n) {
        for(int i =0; i<n.size(); i++)
            if(n[i]==0){
                for(int j = i+1; j<n.size(); j++){
                    if(n[j]!=0){
                        swap(n[i],n[j]);
                        break;
                    }
                }
            }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        j = -1
        for i in range(len(nums)):
            if nums[i]!=0 and i != j and j != -1:
                nums[j] = nums[i]
                nums[i] = 0
                j+=1
            if j == -1 and nums[i]==0: j = i 
        